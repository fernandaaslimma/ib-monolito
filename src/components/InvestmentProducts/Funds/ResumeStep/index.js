import ResumeStep from "./ResumeStep";

export default ResumeStep;
