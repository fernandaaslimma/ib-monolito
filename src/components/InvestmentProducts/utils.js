import { translate } from "../../utils/i18n";

export const verifyDays = (day, type) => {
  if (day == 0) {
    return translate("SAME_DAY_FUNDS");
  } else if (day == 1) {
    return type === "Útil"
      ? translate("ONE_WORKING_DAY_FUNDS")
      : translate("ONE_CONSECUTIVE_DAY_FUNDS");
  } else if (day > 1) {
    return type === "Útil"
      ? `${day}${translate("WORKING_DAYS_FUNDS")}`
      : `${day}${translate("CONSECUTIVE_DAYS_FUNDS")}`;
  }
};
