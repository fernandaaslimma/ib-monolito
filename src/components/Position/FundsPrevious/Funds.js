import React, { Component } from "react";
import { arrayOf, shape, number, string, func, bool } from "prop-types";

import Hide from "../../common/Hide";
import Mobile from "./Mobile";
import Desktop from "./Desktop";
import ErrorBoundary from "../../ErrorBoundary/ErrorBoundary";

class Funds extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isEmpty: false
    };
  }

  async componentDidMount() {
    this.props.getTotalFunds();
    await this.props.getFunds();
    this.setState({ isEmpty: this.props.funds.length === 0 });
  }

  render() {
    const { funds, totalFunds, loading, error } = this.props;
    const { isEmpty } = this.state;

    return (
      <ErrorBoundary errorStatus={error}>
        <div data-test="Position_Funds">
          <Hide above="md">
            <Mobile
              funds={funds}
              totalFunds={totalFunds}
              isEmpty={isEmpty}
              loading={loading}
            />
          </Hide>
          <Hide below="md">
            <Desktop
              funds={funds}
              totalFunds={totalFunds}
              loading={loading}
              isEmpty={isEmpty}
            />
          </Hide>
        </div>
      </ErrorBoundary>
    );
  }
}

Funds.defaultProps = {
  funds: [],
  loading: false
};

Funds.propTypes = {
  funds: arrayOf(
    shape({
      name: string,
      date: string,
      investmentDate: string,
      iofBalance: number,
      grossResultBalance: number,
      grossBalance: number,
      incomeTaxBalance: number,
      portfolioShare: number
    })
  ),
  loading: bool,
  getFunds: func.isRequired,
  getTotalFunds: func.isRequired
};

export default Funds;
