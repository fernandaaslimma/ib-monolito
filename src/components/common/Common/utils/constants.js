export const PT_BR = "pt-BR";
export const EN_US = "en-US";
export const ZH_CN = "zh-CN";
export const SCREEN_LG = 1199;
export const SCREEN_MD = 992;
export const SCREEN_SM = 767;
export const SCREEN_XS = 449;
export const SCREEN_XXS = 319;
export const UP = "up";
export const DOWN = "down";
export const RIGHT = "right";
export const PAGINATION_TOTAL = "X-Total-Count";
export const DEFAULT_MOBILE_HEADER_HEIGHT = 60;
export const ASCENDING = "ASC";
export const DESCENDING = "DESC";
