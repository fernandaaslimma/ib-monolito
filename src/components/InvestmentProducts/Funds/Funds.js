import React, { Component, Fragment } from "react";
import ErrorBoundary from "../../ErrorBoundary/ErrorBoundary";
import { StepSlider } from "react-bocombbm-components";
import DefaultShimmerLoading from "../../common/DefaultShimmerLoading";
import ListStep from "./ListStep";
import DetailsStep from "./DetailsStep";
import FormStep from "./FormStep";
import ConfirmationStep from "./ConfirmationStep";
import ResumeStep from "./ResumeStep";
import moment from "moment";
import {
  getDateFieldPlaceholderByLocale,
  isPtBR,
  translate
} from "../../../utils/i18n";
import { hotjarTag } from "../../../utils/hotjarFun";
import { checkDate } from "../../../utils/validations/fund";
import {
  REGISTRATION_DATA_NOTIFICATION_TYPE,
  SUITABILITY_NOTIFICATION_TYPE,
  FUNDS_SERVICE,
  ACTION_TYPE_APPROVE_INVESTMENT,
  TYPE_PRODUCT_FUNDS,
  TYPE_PRODUCT_FUNDS_UNSUITABLE
} from "../../../utils/constants";
import { unFormatNumber } from "../../../utils/formatNumber";
import { MODAL_TYPES } from "../../common/Modal/Modal";
import Header from "../../common/Modal/Header";
import EFTToken from "../../common/EFTToken";
import { EFTTokenMfaWrapper } from "./styles";

import { InstanceContext } from "./fundsContext";
class InvestmentFunds extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loading: true,
      selectedFund: null,
      typeOfFundPendencie: "",
      route: "",
      filledValue: isPtBR() ? "0,00" : "0.00",
      selectedAccount: {},
      canInvest: false,
      hasTerms: null,
      signed: false,
      hasUnsuitableTerm: [],
      fundsInMoviment: "",
      chosenOperation: null
    };

    this.colors = {
      moderate: "#E3F1D4",
      aggressive: "#E2DCF5",
      conservative: "#DAE6F2"
    };

    this.steps = [
      ListStep,
      DetailsStep,
      FormStep,
      ConfirmationStep,
      ResumeStep
    ];
    this.selectFund = this.selectFund.bind(this);
    this.verifyPendencies = this.verifyPendencies.bind(this);
    this.changeAmmount = this.changeAmmount.bind(this);
    this.selectAccount = this.selectAccount.bind(this);
    this.createOperationFunction = this.createOperationFunction.bind(this);
    this.tokenModal = this.tokenModal.bind(this);
    this.approveTerm = this.approveTerm.bind(this);
    this.resetInvestability = this.resetInvestability.bind(this);
    this.resetFundPendencie = this.resetFundPendencie.bind(this);
    this.downloadFile = this.downloadFile.bind(this);
    this.isMobile = this.isMobile.bind(this);
    this.checkFundIsInMoviment = this.checkFundIsInMoviment.bind(this);
    this.setChosenOperation = this.setChosenOperation.bind(this);
  }

  selectFund(selectedFund, callBack) {
    this.setState({ selectedFund });
    callBack();
  }

  setChosenOperation(operation) {
    this.setState({ chosenOperation: operation });
  }

  selectAccount(event) {
    const selectedAccount = this.props.accounts.find(
      account => account.accountNumber === event.target.value
    );

    this.setState({ selectedAccount });
  }

  changeAmmount(filledValue) {
    this.setState({ filledValue });
  }

  resetInvestability() {
    return this.setState({
      canInvest: false,
      signed: false,
      filledValue: isPtBR() ? "0,00" : "0.00"
    });
  }

  resetFundPendencie() {
    this.setState({ typeOfFundPendencie: "" });
  }

  async componentDidMount() {
    hotjarTag();
    await this.props.handleUserInputTranferDataDate(
      moment().format(getDateFieldPlaceholderByLocale())
    );
    await this.props.getInvestmentFunds({ showClosedFunds: true });
    await this.props.getFunds({ groupBy: "AssetName" });
    await this.props.getAvailableDateRanges(FUNDS_SERVICE);
    await this.props.getAccounts();
    await this.props.getPendingTransactionsFunds();
    this.setState({
      loading: false,
      selectedAccount: this.props.accounts[0]
    });
  }

  componentWillUnmount() {
    hotjarTag();
  }

  showFooterModal(route, typeOfFundPendencie) {
    this.setState({
      typeOfFundPendencie,
      route
    });
  }

  async verifyPendencies() {
    await this.props.getSubscriptionsPendencies();

    if (
      this.props.subscriptionsPendencies &&
      this.props.subscriptionsPendencies.length > 0
    ) {
      this.props.subscriptionsPendencies.every(item => {
        if (item.code === REGISTRATION_DATA_NOTIFICATION_TYPE) {
          this.showFooterModal(
            "/registrationData",
            REGISTRATION_DATA_NOTIFICATION_TYPE
          );
          return false;
        }
        if (item.code === SUITABILITY_NOTIFICATION_TYPE) {
          this.showFooterModal("/suitability", SUITABILITY_NOTIFICATION_TYPE);
          return false;
        }
        return true;
      });
    } else {
      this.setState({
        canInvest: true
      });
    }
  }

  async approveTerm(goToStep) {
    const {
      openToastr,
      closeToastr,
      subscription,
      approveNewTerms
    } = this.props;

    const productTermIdPath = subscription.terms.filter(
      element => element.type === TYPE_PRODUCT_FUNDS
    );

    const unsuitableTermIdPath = subscription.terms.filter(
      element => element.type === TYPE_PRODUCT_FUNDS_UNSUITABLE
    );

    try {
      productTermIdPath.length > 0 &&
        (await approveNewTerms({
          termId: productTermIdPath[0].id,
          accepted: true
        }));

      unsuitableTermIdPath.length > 0 &&
        (await approveNewTerms({
          termId: unsuitableTermIdPath[0].id,
          accepted: true
        }));

      closeToastr();

      this.setState({ signed: true });

      this.tokenModal(goToStep);
    } catch (error) {
      openToastr({
        text: translate("TOASTR_ERROR_ACCEPTED_NEW_TERMS"),
        isBelow: false,
        isTop: true,
        error: true,
        timeout: 600000
      });
    }
  }

  async downloadFile(file, filename, unsuitableTerm) {
    let fileCodeId = file.split(/[/]+/).pop();
    if (this.isMobile()) {
      await this.props.downloadTermsFunds(fileCodeId, unsuitableTerm);
    } else {
      const link = document.createElement("a");
      document.body.appendChild(link);
      link.href = file;
      link.download = filename;
      link.target = "_blank";
      link.click();
      setTimeout(link.remove(), 100);
      return link;
    }
  }

  isMobile() {
    return typeof MOBILEAPP_Download === "function";
  }

  handleClose() {
    this.props.closeModal();
    this.setState({ signed: false });
  }

  async createOperationFunction(goToStep) {
    const { createSubscription } = this.props;
    const {
      filledValue,
      selectedFund,
      selectedAccount: { account, bankCode, branch, number, holderId }
    } = this.state;

    const body = {
      productId: selectedFund.id,
      investmentValue: unFormatNumber(filledValue),
      account: {
        id: account,
        bankCode: bankCode,
        branch: branch,
        number: number
      },
      clientId: holderId
    };
    await createSubscription(body);

    const hasTermToApprove =
      this.props.subscription.terms && this.props.subscription.terms.length > 0;

    const hasUnsuitableTerm =
      this.props.subscription.terms &&
      this.props.subscription.terms.filter(
        element => element.type === TYPE_PRODUCT_FUNDS_UNSUITABLE
      );

    if (hasTermToApprove) {
      this.setState({
        hasUnsuitableTerm: hasUnsuitableTerm.length > 0,
        hasTerms: this.props.subscription
      });
    } else {
      this.tokenModal(goToStep);
    }
  }

  async tokenModal(goToStep) {
    const {
      methodChanging,
      subscription,
      openModal,
      approveSubscription
    } = this.props;
    const subscriptionId = subscription.id;

    const params = {
      actionType: ACTION_TYPE_APPROVE_INVESTMENT,
      payload: {
        id: subscriptionId,
        ...subscription
      }
    };

    const onConfirm = async () => {
      try {
        await approveSubscription(subscriptionId);
        this.props.closeModal();
        goToStep(5);
      } catch (error) {
        this.props.addError(error);
        this.props.closeModal();
        return { error };
      }
    };

    openModal({
      type: MODAL_TYPES.CUSTOM,
      overwriteDefaultButtons: true,
      children: () => (
        <Fragment>
          <Header
            onClickBack={
              methodChanging
                ? () => this.props.changeFactorTogle(false)
                : () => this.handleClose()
            }
            onClickClose={() => this.handleClose()}
            dataTest="MfaAddpBbmBack"
          />
          <EFTTokenMfaWrapper>
            <EFTToken
              onConfirm={onConfirm}
              onMFAConfirmation={() => onConfirm()}
              onMFAError={() => {}}
              createAuthCodeParams={params}
            />
          </EFTTokenMfaWrapper>
        </Fragment>
      )
    });
  }

  async checkFundIsInMoviment(fundName, serverTime) {
    const thisFundToday = this.props.pendingTransactionsFunds.filter(fund => {
      return (
        fund.product.name === fundName &&
        checkDate(fund.transactionDate, serverTime)
      );
    });

    if (thisFundToday.length === 0) {
      this.setState({ fundsInMoviment: "" });
    } else {
      this.setState({ fundsInMoviment: thisFundToday });
    }
  }

  render() {
    const { loading } = this.state;
    return (
      <Fragment>
        {loading ? (
          <DefaultShimmerLoading repeat={4} innerRepeat={2} />
        ) : (
          <ErrorBoundary errorStatus={this.props.error}>
            <InstanceContext.Provider value={this}>
              <StepSlider steps={this.steps} />
            </InstanceContext.Provider>
          </ErrorBoundary>
        )}
      </Fragment>
    );
  }
}

export default InvestmentFunds;
