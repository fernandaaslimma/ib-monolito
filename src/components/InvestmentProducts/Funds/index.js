import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";
import Funds from "./Funds";
import actions from "../../../actions/investmentProducts";
import EFTActions from "../../../actions/EFT";
import MFATokenActions from "../../../actions/mfa";
import modalActions from "../../../actions/modal";
import toastrActions from "../../../actions/toastr";
import termsActions from "../../../actions/terms";
import loginActions from "../../../actions/login";

import positionActions from "../../../actions/position";
import withMaintenanceMode from "../../common/withMaintenanceMode";
import addError from "../../../actions/error";

const Comp = connect(
  ({
    funds,
    getFunds,
    loading,
    error,
    investmentFunds,
    availableDateRanges,
    serverTime,
    subscriptionsPendencies,
    authFactors,
    accounts,
    subscription,
    openToastr,
    closeToastr,
    userInfo,
    pendingTransactionsFunds
  }) => ({
    funds,
    getFunds,
    loading,
    error,
    investmentFunds,
    availableDateRanges,
    serverTime,
    subscriptionsPendencies,
    authFactors,
    accounts,
    subscription,
    openToastr,
    closeToastr,
    userInfo,
    pendingTransactionsFunds
  }),
  combineActions(
    actions,
    EFTActions,
    positionActions,
    MFATokenActions,
    modalActions,
    toastrActions,
    addError,
    termsActions,
    loginActions
  )
)(Funds);

export default withMaintenanceMode(Comp);
