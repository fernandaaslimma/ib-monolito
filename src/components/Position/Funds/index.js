import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";
import Funds from "./Funds";
import actions from "../../../actions/investmentProducts";
import EFTActions from "../../../actions/EFT";
import MFATokenActions from "../../../actions/mfa";
import modalActions from "../../../actions/modal";
import toastrActions from "../../../actions/toastr";
import termsActions from "../../../actions/terms";

import positionActions from "../../../actions/position";
import withMaintenanceMode from "../../common/withMaintenanceMode";
import addError from "../../../actions/error";

const Comp = connect(
  ({
    funds,
    totalFunds,
    getFunds,
    getTotalFunds,
    loading,
    error,
    investmentFunds,
    availableDateRanges,
    serverTime,
    subscriptionsPendencies,
    authFactors,
    accounts,
    subscription,
    redemption,
    openToastr,
    closeToastr,
    pendingTransactionsFunds,
    userInfo
  }) => ({
    funds,
    totalFunds,
    getFunds,
    getTotalFunds,
    loading,
    error,
    investmentFunds,
    availableDateRanges,
    serverTime,
    subscriptionsPendencies,
    authFactors,
    accounts,
    subscription,
    redemption,
    openToastr,
    closeToastr,
    pendingTransactionsFunds,
    userInfo
  }),
  combineActions(
    actions,
    EFTActions,
    positionActions,
    MFATokenActions,
    modalActions,
    toastrActions,
    addError,
    termsActions
  )
)(Funds);

export default withMaintenanceMode(Comp);
