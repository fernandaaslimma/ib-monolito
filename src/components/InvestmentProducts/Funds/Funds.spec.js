import React from "react";
import { shallow } from "enzyme";
import Funds from "./Funds";
import { TYPE_PRODUCT_FUNDS_UNSUITABLE } from "../../../utils/constants";
import { isPtBR } from "../../../utils/i18n";

const fundsData = [
  {
    id: 1,
    name: "BOCOM BBM CORPORATE CREDIT FIC MULTIMERCADO CRÉDITO PRIVADO", // IP Atlas USD FIC FIA IE
    cnpj: "11111111111",
    description: "descrição do fundo...",
    manager: "BOCOM BBM", // Gestor
    administrator: "Bradesco", // Administrador
    minimumBalance: 20000.0, // Saldo mínimo
    minimumTransaction: 10000.0, // Movimentação mín.
    initialInvestment: 20000.0, // Aplicação inicial de
    riskProfile: "Aggressive",
    riskProfileLabel: "Agressivo", // Agressivo
    returns: {
      twelveMonths: 0.1, // Rentab. Últimos 12 meses de
      year: 0.1, // Ano
      thirtySixMonths: 0.1, // 36 meses
      fortyEightMonths: 0.1 // 48 meses
    },
    class: "HedgeFund",
    classLabel: "Multimercado", // Renda fixa
    subscription: {
      type: "corrido",
      conversionDays: 0, // Prazo para conversão das cotas na aplicação ?
      settlementDays: 0 // Valor na conta (pós conversão das cotas) ?
    },
    redemption: {
      type: "corrido",
      conversionDays: 60, // Prazo para conversão das cotas na aplicação ?
      settlementDays: 1 // Valor na conta (pós conversão das cotas) ?
    },
    benchmark: "CDI", // Benckmark
    quotaDate: "2020-12-31",
    administrationFee: 0.2, // Taxa de administração
    performanceFee: 0.2, // Taxa de perfomance
    inceptionDate: "2020-12-31" // Data de ínicio do fundo
  }
];

const props = {
  getInvestmentFunds: jest.fn(() => Promise.resolve(fundsData)),
  handleUserInputTranferDataDate: jest.fn(),
  getAvailableDateRanges: jest.fn(),
  getSubscriptionsPendencies: jest.fn(),
  getAccounts: jest.fn(),
  createSubscription: jest.fn(() => Promise.resolve("mock")),
  UserInputTranferDataDate: jest.fn(),
  changeFactorTogle: jest.fn(),
  approveNewTerms: jest.fn(),
  closeToastr: jest.fn(),
  closeModal: jest.fn(),
  openToastr: jest.fn(),
  methodChanging: true,
  openModal(config) {
    config.children();
  },
  subscriptionsPendencies: [
    {
      document: "Ficha Cadastral"
    },
    {
      document: "Questionário de Suitability"
    }
  ],
  accounts: [
    {
      account: 1,
      accountNumber: "107 2 4-3",
      blockedBalance: 7,
      availableBalance: 6,
      totalBalance: 5,
      date: "12/12",
      document: "",
      name: "",
      bankISPB: "2312321",
      verifyingDigit: 3,
      number: 4,
      branch: 2,
      bankCode: 107
    }
  ],
  subscription: {
    id: 145,
    productId: 1,
    idempotencyKey: "2b92510a-40e6-4a1e-9857-4741c5852e62",
    investmentValue: 40000.0,
    status: "AWAITING_APPROVAL",
    terms: [
      {
        name: "Term of adhesion BOCOM BBM CORPORATE CREDIT 60 FIC FIM CP",
        type: "productTerms",
        url:
          "https://api.bocombbm.com.br/productterms/fundos/adesao_37322143000120.pdf",
        id: 48
      },
      {
        name: "Non Compliance Term - Fixed Income",
        type: TYPE_PRODUCT_FUNDS_UNSUITABLE,
        url:
          "https://api.bocombbm.com.br/noncomplianceterms/fundos/desenquadramento_rf_mod.pdf",
        id: 250
      }
    ]
  }
};

describe("Funds component", () => {
  it("should match snapshot with loading", () => {
    expect(shallow(<Funds />)).toMatchSnapshot();
  });

  it("should match snapshot with data", () => {
    expect(shallow(<Funds {...props} />)).toMatchSnapshot();
  });

  it("should select a fund", () => {
    const spy = jest.fn();
    const wrapperInstance = shallow(<Funds {...props} />).instance();
    const spyState = jest.spyOn(wrapperInstance, "setState");
    wrapperInstance.selectFund("mock", spy);

    expect(spyState).toHaveBeenCalledWith({ selectedFund: "mock" });
    expect(spy).toHaveBeenCalled();
  });

  it("should call verifyPendencies with Ficha Cadastral", () => {
    const wrapperInstance = shallow(<Funds {...props} />).instance();
    wrapperInstance.verifyPendencies();

    expect(props.getSubscriptionsPendencies).toHaveBeenCalled();
  });

  it("should call verifyPendencies with Suitability", () => {
    let newProps = {
      ...props,
      subscriptionsPendencies: [
        {
          document: "Questionário de Suitability"
        }
      ]
    };
    const wrapperInstance = shallow(<Funds {...newProps} />).instance();
    wrapperInstance.verifyPendencies();

    expect(props.getSubscriptionsPendencies).toHaveBeenCalled();
  });

  it("should call verifyPendencies with Ficha Patrimonial", () => {
    let newProps = {
      ...props,
      subscriptionsPendencies: [
        {
          document: "Ficha Patrimonial"
        }
      ]
    };
    const wrapperInstance = shallow(<Funds {...newProps} />).instance();
    wrapperInstance.verifyPendencies();

    expect(props.getSubscriptionsPendencies).toHaveBeenCalled();
  });

  it("should call verifyPendencies withou pendencies", () => {
    let newProps = { ...props, subscriptionsPendencies: [] };
    const wrapperInstance = shallow(<Funds {...newProps} />).instance();

    wrapperInstance.verifyPendencies();

    expect(props.getSubscriptionsPendencies).toHaveBeenCalled();
  });

  it("should call verifyPendencies and show person registration modal", () => {
    let newProps = {
      ...props,
      subscriptionsPendencies: [{ code: "PersonRegistrationForms" }]
    };
    const wrapperInstance = shallow(<Funds {...newProps} />).instance();
    wrapperInstance.verifyPendencies();
  });

  it("should call verifyPendencies and show suitability modal", () => {
    let newProps = {
      ...props,
      subscriptionsPendencies: [{ code: "SuitabilityForms" }]
    };
    const wrapperInstance = shallow(<Funds {...newProps} />).instance();
    wrapperInstance.verifyPendencies();
  });

  it("should call resetInvestability", () => {
    const wrapperInstance = shallow(<Funds {...props} />).instance();

    const spy = jest.spyOn(wrapperInstance, "setState");
    wrapperInstance.resetInvestability();

    expect(spy).toHaveBeenCalledWith({
      canInvest: false,
      signed: false,
      filledValue: isPtBR() ? "0,00" : "0.00"
    });
  });

  it("should call resetFundPendencie", () => {
    const wrapperInstance = shallow(<Funds {...props} />).instance();

    const spy = jest.spyOn(wrapperInstance, "setState");
    wrapperInstance.resetFundPendencie();

    expect(spy).toHaveBeenCalledWith({
      typeOfFundPendencie: ""
    });
  });

  it("should call handleClose", () => {
    const wrapperInstance = shallow(<Funds {...props} />).instance();

    const spy = jest.spyOn(wrapperInstance, "setState");
    wrapperInstance.handleClose();

    expect(props.closeModal).toHaveBeenCalled();

    expect(spy).toHaveBeenCalledWith({
      signed: false
    });
  });

  it("should select an account", () => {
    const wrapperInstance = shallow(<Funds {...props} />).instance();

    const spy = jest.spyOn(wrapperInstance, "setState");
    wrapperInstance.selectAccount({ target: { value: "107 2 4-3" } });

    expect(spy).toHaveBeenCalled();
  });

  it("should change the ammount", () => {
    const wrapperInstance = shallow(<Funds {...props} />).instance();

    const spy = jest.spyOn(wrapperInstance, "setState");
    wrapperInstance.changeAmmount("100.000,00");

    expect(spy).toHaveBeenCalledWith({ filledValue: "100.000,00" });
  });

  it("should show footer modal", () => {
    const wrapperInstance = shallow(<Funds {...props} />).instance();

    const spy = jest.spyOn(wrapperInstance, "setState");
    wrapperInstance.showFooterModal("route", "pendency");

    expect(spy).toHaveBeenCalledWith({
      typeOfFundPendencie: "pendency",
      route: "route"
    });
  });

  it("should create subscription and change the state hasTerms", () => {
    const wrapperComp = shallow(<Funds {...props} />);

    wrapperComp.setState({ selectedFund: { id: "123" } });
    wrapperComp.update();

    const wrapperInstance = wrapperComp.instance();

    wrapperInstance.createOperationFunction("mockfunc");
  });

  it("should create subscription and invoke token modal", () => {
    const newProps = { ...props, subscription: {} };
    const wrapperComp = shallow(<Funds {...newProps} />);
    wrapperComp.setState({ selectedFund: { id: "123" } });
    wrapperComp.update();

    const wrapperInstance = wrapperComp.instance();
    wrapperInstance.createOperationFunction("mockfunc");
  });

  it("should call approveTerms", () => {
    const wrapperComp = shallow(<Funds {...props} />);

    const wrapperInstance = wrapperComp.instance();

    wrapperInstance.approveTerm(jest.fn());
  });
});
