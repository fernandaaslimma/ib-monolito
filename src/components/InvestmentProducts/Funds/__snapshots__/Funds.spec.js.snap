// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Funds component should match snapshot with data 1`] = `
<Fragment>
  <DefaultShimmerLoading
    innerRepeat={2}
    repeat={4}
  />
</Fragment>
`;

exports[`Funds component should match snapshot with loading 1`] = `
<Fragment>
  <DefaultShimmerLoading
    innerRepeat={2}
    repeat={4}
  />
</Fragment>
`;
